types/project.go

```diff
@@ -407,25 +433,26 @@ func IgnoreDependencies(options *withServicesOptions) {
 	options.dependencyPolicy = ignoreDependencies
 }
 
-// ForServices restrict the project model to selected services and dependencies
-func (p *Project) ForServices(names []string, options ...DependencyOption) error {
+// ForServices restricts the project model to selected services and dependencies
+// It returns a new Project instance with the changes and keep the original Project unchanged
+func (p *Project) ForServices(names []string, options ...DependencyOption) (*Project, error) {
```

Comment:
(see other comment re: getting rid of `error` from `deepCopy()`)

I am wondering if it makes sense to ignore unknown names here, so that this method can return `*Project`, which would make it much more ergonomic to use: `p = p.ForServices(svc)`.

In particular, this could allow chaining, e.g.
```go
p.ForServices(svcs).ForEach(...)
```

Reply:
Or we can introduce an `Either` type 😇 , WDYT?

Reply:
> I am wondering if it makes sense to ignore unknown names here

this would be a bug then if we silently ignore incorrect parameter being set